Automattic/jetpack
# My Jetpack: Fix Jetpack AI interstitial on disconnected site

## Proposed changes:

* Show the 100 requests tier interstitial when the site is disconnected, so we can sell the plan even before the connection
* To do it:
   * change the backend to default to 0 as current tier and 100 as next tier when the site is not connected
   * change the client code to check the connection status and default the next tier quantity to 100 when the site is not registered/connected
* Prevent remote requests from the Jetpack AI product class inside My Jetpack, bailing early when the site is not connected

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* You need a disconnected site to test it; you can start with a non-connected, fresh JN site, or disconnect your usual testing site
* After confirming the site is disconnected, go to the My Jetpack page
* Scroll down looking for the Jetpack AI card
* It should show up with a "Learn more" call-to-action:

<img width="400" alt="Screenshot 2024-01-02 at 19 00 42" src="https://github.com/Automattic/jetpack/assets/6760046/142c535b-c7dc-4e44-a1c2-e522e1b029f8">

* Click the "Learn more" button; It should open the Jetpack AI product interstitial
* **Before** this change, the interstitial would show the "Contact Us" messaging:

<img width="600" alt="Screenshot 2024-01-02 at 19 04 19" src="https://github.com/Automattic/jetpack/assets/6760046/b729ea74-f9cf-46e0-920d-af007603dede">

* **After** this change, the interstitial will show the "100 requests" tier messaging:

<img width="600" alt="Screenshot 2024-01-03 at 17 35 49" src="https://github.com/Automattic/jetpack/assets/6760046/cbd8d9c0-226b-4c5f-8fee-eb7893aad036">

* You should NOT see the price for the product, since prices are a remote information
* Clicking the "Get Jetpack AI" button should lead you to the checkout page, with the 100 requests tier selected for purchase:

<img width="400" alt="Screenshot 2024-01-02 at 18 23 51" src="https://github.com/Automattic/jetpack/assets/6760046/944ce4f6-eee4-4e26-a98d-8c5a6ba8d159">
